db/db.go

```diff
@@ -280,26 +302,34 @@ func (d *DB) MakeAlias(ctx context.Context, uid string, path string) (string, er
 
 	// get the mapping collection
 	col := d.Collection(path)
-	// get the snapshot of the document with the requested wid
-	snap, err := col.Doc(wid).Get(ctx)
 
-	//if the doc id is taken, generate a different wid
-	for snap.Exists() == true {
-
-		aid++
-		if aid >= 0xFFFFFFFFF {
-			aid = 0
+	err := d.RunTransaction(ctx, func(ctx context.Context, tx *firestore.Transaction) error {
+		doc, err := tx.Get(col.Doc(wid))
+		if err != nil {
+			return err
 		}
```

Comment:
The firestore library spec states that err will be set if the document doesn't exist. In this case, we are expecting the document not to exist, so one would expect err to contain a message complaining about the document with their supplied rpc code `NotFound`. Indeed, this is the case, so the function was returning early with an error that we expected, which fails the tests. This same error occurs L324..326

My quick suggestion as to how to fix this is to be more specific in the error checks here. Something to the tune of `doc == nil || (err != nil && doc.Exists())` might do the trick, checking if we have an error despite the document existing.

Aside from this, LGTM!